services/beaconchain/src/main/java/tech/pegasys/teku/services/beaconchain/WeakSubjectivityInitializer.java

```diff
@@ -141,6 +141,15 @@ public SafeFuture<WeakSubjectivityConfig> finalizeAndStoreConfig(
 
   public void validateInitialAnchor(
       final AnchorPoint initialAnchor, final UInt64 currentSlot, final Spec spec) {
+    final Fork expectedFork = spec.getForkSchedule().getFork(initialAnchor.getEpoch());
+    final Fork loadedFork = initialAnchor.getState().getFork();
+    if (!isSameForkInfo(expectedFork, loadedFork)) {
+      throw new InvalidConfigurationException(
+          String.format(
+              "The fork from the initial-state (%s) does not match the configured fork schedule (%s).\nPlease check that network in configuration matches the loaded state.",
+              loadedFork, expectedFork));
+    }
+
```

Comment:
NIT: just for readability this could moved to a method like `verifyInitialStateFork(..)` or something like that.

Reply:
i had a quick look, it needs a bigger rewrite that i dont have adequate time to do and test, so not actioning this.